devenv/lib/archive.py

```diff
@@ -77,22 +87,63 @@ def download(
 
 
 # mutates members!
-# strips the leading component (/ is always stripped and doesn't count)
-# and optionally replaces with a new prefix
-def strip1(members: Sequence[tarfile.TarInfo], new_prefix: str = "") -> None:
+# strips N leading components and optionally adds a new prefix
+# (what ends up being stripped should be a common prefix for all entries)
+# (/ is always stripped and doesn't count)
+def strip_n(
```

Comment:
without rewriting `strip1` -- can't `strip_n` just be a composition of `n` `strip1`s?

Reply:
i guess it can, but i'm so proud of my unnecessary optimization